spec/support/ar_classes.rb

```diff
@@ -1,12 +1,35 @@
-ActiveRecord::Base.establish_connection(
-  :adapter => 'sqlite3',
-  :database => ':memory:'
-)
+def establish_active_record_connection
+  ActiveRecord::Base.establish_connection(
+    :adapter => 'sqlite3',
+    :database => ':memory:'
+  )
+
+  [:MockableModel,
+   :SubMockableModel,
+   :AssociatedModel,
+   :AlternatePrimaryKeyModel].each do |host|
+    next unless Kernel.const_defined? host
+    Connections.extended(Kernel.const_get(host))
+  end
+end
+
+def clear_active_record_connection
+  ActiveRecord::Base.connection_handler.clear_all_connections!
+  ActiveRecord::Base.connection_handler.connection_pool_list.each do |con|
+    if ::Rails::VERSION::STRING > '5'
+      ActiveRecord::Base.connection_handler.remove_connection(con.spec.name)
+    elsif ::Rails::VERSION::STRING > '3'
+      ActiveRecord::Base.connection_handler.remove_connection(ActiveRecord::Base)
+    end
+  end
+end
+
+establish_active_record_connection
 
 module Connections
   def self.extended(host)
     host.connection.execute <<-eosql
-      CREATE TABLE #{host.table_name} (
+      CREATE TABLE IF NOT EXISTS #{host.table_name} (
```

Comment:
When would this exist?

Reply:
It has been removed. Marked as *resolved*.